inference-engine/src/gna_plugin/backend/gna_limitations.cpp

```diff
@@ -182,27 +183,195 @@ bool AreLayersSupported(InferenceEngine::CNNNetwork& network, std::string& errMe
+                                                       // Skip all convolutions in this check, they will be handled during concat primitive creation
+                                                       auto isFusableWithConv = [](InferenceEngine::CNNLayerPtr ptr) {
+                                                           return (LayerInfo(ptr).isFusableWithConv() || LayerInfo(ptr).isNonFunctional() ||
+                                                               (LayerInfo(ptr).isPermute() && ptr->input()->getLayout() == InferenceEngine::Layout::NCHW &&
+                                                                   ptr->GetParamAsInts("order") ==
+                                                                   GetPermuteOrder(InferenceEngine::Layout::NCHW, InferenceEngine::Layout::NHWC)));
+                                                       };
+
+                                                       for (auto input_idx = 0; input_idx != concat_layer->insData.size(); input_idx++) {
+                                                           prev_layer = InferenceEngine::CNNNetPrevLayerSkipCertain(layer, input_idx, isFusableWithConv);
+                                                           if (prev_layer && LayerInfo(prev_layer).isConvolution())
+                                                               return;
+                                                       }
+
+                                                       // Look for trivial cases which will be flattened later
+                                                       bool is_not_trivial_concat = false;
+
+                                                       // All const inputs case is supported
+                                                       bool concat_all_const_inputs = false;
+
+                                                       // If concat axis > 0, detect any dimension > 1 before the concat axis
+                                                       if (concat_axis > 0) {
+                                                           for (auto axis = 0; axis < concat_axis; axis++) {
+                                                               if (in_dims[axis] > 1) {
+                                                                   is_not_trivial_concat = true;
+                                                                   break;
+                                                               }
+                                                           }
+                                                       // If concat axis == 0, detect any preceding functional layer's input
+                                                       // with 0'th dimension > 1, but take into account that some layers need to be skipped
+                                                       } else {
+                                                           concat_all_const_inputs = true;
```

Comment:
smoke_WeighableLayerWithoutFqTest fails without this check, but it uses const and and an input with first dimension > 1. I'll change the variable name to reflect this better, because it allows such scenario also.